src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int OFFSET = 1;
+    private static final int SIZE_ADJUSTMENT = 1;
+    private static final int START_INDEX = 0;
+    private static final double ELEMENT_DATA_MULTIPLIER = 1.5;
+    private int listCapacity = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[listCapacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == listCapacity) {
+            grow();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < START_INDEX) {
+            checkIndex(index);
```

Comment:
There's no need to call `checkIndex(index)` if the index is greater than the size or less than START_INDEX since it will throw an exception in the checkIndex method anyway. You should remove this if statement.